lib/fetch/index.js

```diff
@@ -2039,7 +2033,7 @@ async function httpNetworkFetch (
 
       // 9. If stream doesn’t need more data ask the user agent to suspend
       // the ongoing fetch.
-      if (!fetchParams.controller.controller.desiredSize) {
+      if (fetchParams.controller.controller.desiredSize <= 0) {
```

Comment:
According to MDN desiredSize can go negative if the stream is oversaturized. !(-1) is false. And if desiredSize is 0, then we know we can not push data, anyway. 0 would actually mean, that the stream is closed. So is my change correct?